### PR TITLE
Extract config from application.rb to initializers

### PR DESCRIPTION
**Pull request by dcyoung-dev**
## Context

When upgrading Rails versions, the `rails app:update` generator often overwrites the `application.rb` file. 
This can be confusing as it requires time and effort to step through each change and validate the need for the change. 

Moving configuration out into separate initializers reduces the number of changes required to the `application.rb`. 
This also groups config together to make it easier to find and change when needed. 

## Changes proposed in this pull request

Move config out of application.rb to initializers

## Guidance to review

- Thorough click around testing required

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)
- [ ] Inform data insights team due to database changes
- [ ] Make sure all information from the Trello card is in here
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally
- [ ] Add PR link to Trello card

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -47,77 +37,14 @@ class Application < Rails::Application
...
-
-    config.middleware.insert_before Rack::Sendfile, RedirectToServiceGovUkMiddleware
-    config.middleware.use RequestIdentityMiddleware
-    config.middleware.use ServiceUnavailableMiddleware
-    config.middleware.use VendorAPIRequestMiddleware
-    config.middleware.use Grover::Middleware
-    config.skylight.probes += %w(redis)
-    config.skylight.environments = ENV["SKYLIGHT_ENABLE"].to_s == "true" ? [Rails.env] : []
-
     config.after_initialize do |app|
```
Comment (dcyoung-dev) [inline config/application.rb:46]: I don't know why this is here and not at the top of the `routes.rb` file

File: config/environments/production.rb [hunk 2]
```diff
@@ -174,12 +174,6 @@ def call(env)
                                              [
                                                ActionDispatch::RemoteIp::TRUSTED_PROXIES,
                                              ]
-                                           else
```
Comment (dcyoung-dev) [inline config/environments/production.rb:176]: We'll check the assumption that we don't need this with Infra before committing to this

File: config/initializers/def_analytics.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Rails.application.configure do
```
Comment (elceebee) [inline config/initializers/def_analytics.rb:1]: Should this file be `dfe_analytics`? Not `def_analytics`?

**Review by elceebee — APPROVED** [review]
I've got no issue with this approach, should be easier to do upgrades and manage config in the future.

**Review by CatalinVoineag — APPROVED** [review]
Looks good apart from the comments already raised

File: config/initializers/action_mailer.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+require './app/lib/hosting_environment'
```
[the remark this answers is not in the document]
Comment (dcyoung-dev) [reply config/initializers/action_mailer.rb:1]: Fixed in https://github.com/DFE-Digital/apply-for-teacher-training/pull/9964/commits/feb7c8850170d30d5c7f22ede92d3908c96f28ce

Had to add the mailer previews to the `autoload`

[the remark this answers is not in the document]
Comment (dcyoung-dev) [reply config/initializers/action_mailer.rb:1]: The path doesn't load 🤔
https://apply-review-9964.test.teacherservices.cloud/rails/mailers/candidate_mailer/chase_reference

[the remark this answers is not in the document]
Comment (dcyoung-dev) [reply config/initializers/action_mailer.rb:1]: In fairness these are not working in other review apps 🤦‍♂️
https://apply-review-9969.test.teacherservices.cloud/rails/mailers/candidate_mailer/chase_reference

[the remark this answers is not in the document]
Comment (dcyoung-dev) [reply config/initializers/action_mailer.rb:1]: Also doesn't work on QA
https://qa.apply-for-teacher-training.service.gov.uk/rails/mailers/candidate_mailer/chase_reference

**Comment by dcyoung-dev** [comment]
[Move config out of `application.rb` into separate initializers](https://trello.com/c/k3cKXd4B/360-move-config-out-of-applicationrb-into-separate-initializers)

**Pull request merged**